src/main/java/edu/kis/powp/jobs2d/events/SelectOpenFromJsonOptionListener.java

```diff
@@ -0,0 +1,51 @@
+package edu.kis.powp.jobs2d.events;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.stream.JsonReader;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.manager.DriverCommandManager;
+import edu.kis.powp.jobs2d.command.manager.InterfaceAdapter;
+import edu.kis.powp.jobs2d.features.CommandsFeature;
+import javax.swing.JFileChooser;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.io.*;
+import java.util.Arrays;
+import java.util.List;
+
+public class SelectOpenFromJsonOptionListener implements ActionListener {
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        JFileChooser chooser = new JFileChooser();
+        File workingDirectory = new File(System.getProperty("user.dir"));
+        chooser.setCurrentDirectory(workingDirectory);
+        if (chooser.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+            File f = chooser.getSelectedFile();
+            System.out.println(f.getName());
+            drawFromJson(f.getPath());
```

Comment:
what if I would like to change json to xml format or read from string rather than a file? Your solution is hardly extensible. You may use Reader interface to be independent in the latter aspect (fix it), but there is no abstraction on the input format level. Consider this:
- input (text, or a Reader)
- output (DriverCommand collection or ICompoudCommand)
You should provide a simple (one method) interface for this purpose.